catroid/src/main/AndroidManifest.xml

```diff
@@ -124,34 +126,28 @@
                 <category android:name="android.intent.category.BROWSABLE" />
                 <data
                     android:host="*"
+                    android:mimeType="application/octet-stream"
                     android:pathPattern="@string/catroid_extension_pathPattern"
-                    android:scheme="http" />
+                    android:scheme="file" />
             </intent-filter>
             <intent-filter>
                 <action android:name="android.intent.action.VIEW" />
                 <category android:name="android.intent.category.DEFAULT" />
                 <category android:name="android.intent.category.BROWSABLE" />
                 <data
                     android:host="*"
+                    android:mimeType="application/octet-stream"
                     android:pathPattern="@string/catroid_extension_pathPattern"
```

Comment:
this is not working at all on my phone, reading up a bit on android:pathPattern i found that:
1. the filename isnt necessarily available to you when you get an URI, so it also doesnt match depending on which application you use to open (download and filemanager on android 26 is not working at all)

2. pathPattern is not performing well with subdirectories, so this will probably only match *.catrobat files in the root directory, this leads to a lot of people doing weird stuff like:
                <data android:pathPattern="/.*\\.catrobat" />
                <data android:pathPattern="/.*\\..*\\.catrobat" />
                <data android:pathPattern="/.*\\..*\\..*\\.catrobat" />
                <data android:pathPattern="/.*\\..*\\..*\\..*\\.catrobat" />
                <data android:pathPattern="/.*\\..*\\..*\\..*\\..*\\.catrobat" />

Reply:
github is mangling the regex...

Reply:
Hm so what can we do here?
If option 2 doesn't work with android 26 apps, do we leave it out completely?